rollout/trafficrouting/istio/controller.go

```diff
@@ -174,32 +174,67 @@ func (c *IstioController) EnqueueRolloutFromIstioVirtualService(vsvc interface{}
 	}
 }
 
+func (c *IstioController) validateRolloutVirtualServicesConfig(r *v1alpha1.Rollout) error {
+	//Either VirtualService or VirtualServices must be configured.
+	//If both configured then it is an invalid configuration
+	errorString := "either VirtualService or VirtualServices must be configured"
+	if istioutil.MultipleVirtualServiceConfigured(r) {
+		if r.Spec.Strategy.Canary.TrafficRouting.Istio.VirtualService.Name != "" {
+			return fmt.Errorf(errorString)
+		}
+	} else {
+		if r.Spec.Strategy.Canary.TrafficRouting.Istio.VirtualService.Name == "" {
+			return fmt.Errorf(errorString)
+		}
+	}
+	return nil
+}
+
 func (c *IstioController) GetReferencedVirtualServices(ro *v1alpha1.Rollout) (*[]unstructured.Unstructured, error) {
+	var fldPath *field.Path
+	fldPath = field.NewPath("spec", "strategy", "canary", "trafficRouting", "istio")
 	ctx := context.TODO()
 	virtualServices := []unstructured.Unstructured{}
-	fldPath := field.NewPath("spec", "strategy", "canary", "trafficRouting", "istio", "virtualService", "name")
 	if ro.Spec.Strategy.Canary != nil {
 		canary := ro.Spec.Strategy.Canary
 		if canary.TrafficRouting != nil && canary.TrafficRouting.Istio != nil {
 			var vsvc *unstructured.Unstructured
 			var err error
-			vsvcNamespace, vsvcName := istioutil.GetVirtualServiceNamespaceName(canary.TrafficRouting.Istio.VirtualService.Name)
-			if vsvcNamespace == "" {
-				vsvcNamespace = ro.Namespace
+			var vsvcs []v1alpha1.IstioVirtualService
+
+			err = c.validateRolloutVirtualServicesConfig(ro)
```

Comment:
As mentioned earlier, this validation should be moved to the earlier validation path.

Reply:
GetReferencedVirtualServices method iterates to rollout virtualServices and stores the k8s virtualService Record.
It makes sense to validate here before proceeding to the rollout virtualServices iteration.

Reply:
Please see my comment about where to put the validation: https://github.com/argoproj/argo-rollouts/pull/1381#discussion_r682662879